### PR TITLE
Add docker publish

### PR DESCRIPTION
**Pull request by mloubout**
Bulid and publish docker images to dockerhub/devitocodes at:
- PR mege into `devitocodes/devito:cpu-dev`
- release into `devitocodes/devito:cpu-v***`

GPU to be added once #1364  is merged

Tested in my fork and seems to work.

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Check event name
+        run: echo ${{ github.event_name }}
+
+      - name: CPU image
+        if: github.event_name == 'push'
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          repository: mloubout/devito
```
Comment (FabioLuporini) [inline .github/workflows/docker-publish.yml:26]: does this need to be your workspace?

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          dockerfile: ./docker/Dockerfile
```
Comment (FabioLuporini) [inline .github/workflows/docker-publish.yml:27]: eventually, we'll add another one for Dockerfile.nvidia right?

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (mloubout) [reply .github/workflows/docker-publish.yml:26]: Oups, debug leftover should be `devitocodes`

File: .github/workflows/docker-publish.yml (hunk 2 above)
Comment (mloubout) [reply .github/workflows/docker-publish.yml:27]: yep the one fro #1364 will be added

Comment (mloubout) [reply .github/workflows/docker-publish.yml:28]: added

**Pull request merged**